src/core/src/types/arrays.rs

```diff
@@ -73,28 +59,13 @@ impl<T: Zero + Clone> Array2<T> {
     /// a.resize_if_different((8, 9));
     /// assert_eq!(a[(3, 3)], 0.0);
     /// ```
-    pub fn resize_if_different(&mut self, size: (Ix, Ix)) {
-        if self.0.shape() != &[size.0, size.1] {
+    pub fn resize_if_different(&mut self, size: (usize, usize)) {
```

Comment:
I agree with you, but I can't think of a better name right now.

Giving this function the `std::vector::resize` behaviour could also be a possibility, by round-triping to a `Vec`:

```
let vec = self.into_vec();
vec.resize(size.0 * size.1);
*self = Array::from_vec(vec, size);
```

I don't know whether we should rename this function or change the behaviour. Do you have any preference?